src/main/java/mate/academy/bookstore/service/CategoryServiceImpl.java

```diff
@@ -0,0 +1,54 @@
+package mate.academy.bookstore.service;
+
+import java.util.List;
+import lombok.RequiredArgsConstructor;
+import mate.academy.bookstore.dto.category.CategoryDto;
+import mate.academy.bookstore.dto.category.CategoryRequestDto;
+import mate.academy.bookstore.excepion.EntityNotFoundException;
+import mate.academy.bookstore.mapper.CategoryMapper;
+import mate.academy.bookstore.model.Category;
+import mate.academy.bookstore.repository.category.CategoryRepository;
+import org.springframework.stereotype.Service;
+
+@Service
+@RequiredArgsConstructor
+public class CategoryServiceImpl implements CategoryService {
+    private final CategoryRepository categoryRepository;
+    private final CategoryMapper categoryMapper;
+
+    @Override
+    public List<CategoryDto> findAll() {
+        return categoryRepository.findAll().stream()
+                .map(categoryMapper::toDto)
+                .toList();
+    }
+
+    @Override
+    public CategoryDto getById(Long id) {
+        Category category = categoryRepository.findById(id).orElseThrow(
+                () -> new EntityNotFoundException("Can't find category by id: " + id)
+        );
+        return categoryMapper.toDto(category);
+    }
+
+    @Override
+    public CategoryDto save(CategoryRequestDto requestDto) {
+        Category category = categoryMapper.toModel(requestDto);
+        return categoryMapper.toDto(categoryRepository.save(category));
+    }
+
+    @Override
+    public CategoryDto update(Long id, CategoryRequestDto requestDto) {
+        Category category = categoryRepository.findById(id).orElseThrow(
+                () -> new EntityNotFoundException("Can't find category by id: " + id)
+        );
+        Category categoryWithId = categoryMapper.toModel(requestDto);
```

Comment:
You can create in CategoryMapper method: 
`void updateCategoryFromDto(CategoryRequestDto requestDto, @MappingTarget Category category);`
and use it here